components/script/dom/validation.rs

```diff
@@ -1,5 +1,10 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use dom::element::Element;
+use dom::validitystate::ValidityStatus;
 
-pub trait Validatable {}
+pub trait Validatable {
+    fn as_element_validatable(&self) -> &Element;
```

Comment:
This method does not belong in this trait. We want a method on Element like [as_maybe_activatable](https://dxr.mozilla.org/servo/rev/639bb3f5a6aaec5c76bcc2a2ea6e71722f91df0d/components/script/dom/element.rs#2265) that returns an instance of the `Validatable` trait. This will allow us to invoke methods via that trait.